src/hera/workflows/script.py

```diff
@@ -358,11 +359,30 @@ def append_annotation(annotation: Union[Artifact, Parameter]):
                 annotation.value_from = ValueFrom(path=outputs_directory + f"/parameters/{annotation.name}")
             parameters.append(annotation)
 
-    if get_origin(inspect.signature(source).return_annotation) is Annotated:
-        append_annotation(get_args(inspect.signature(source).return_annotation)[1])
-    elif get_origin(inspect.signature(source).return_annotation) is tuple:
-        for annotation in get_args(inspect.signature(source).return_annotation):
+    return_annotation = inspect.signature(source).return_annotation
+    if get_origin(return_annotation) is Annotated:
+        append_annotation(get_args(return_annotation)[1])
+    elif get_origin(return_annotation) is tuple:
+        for annotation in get_args(return_annotation):
+            if isinstance(annotation, type) and issubclass(annotation, RunnerOutput):
+                raise ValueError("RunnerOutput cannot be part of a tuple output")
```

Comment:
Only one `RunnerOutput` is allowed as the function return type. Note that we also haven't dealt with nested `Annotated` types above.